benchmarks/GraphBLAS-sharp.Benchmarks/Utils.fs

```diff
@@ -64,44 +82,87 @@ module Utils =
+        |> Array.sortBy (fun struct(packedIndex, _) -> packedIndex)
         |>
-            fun (rows, cols, values) ->
-                let c f x y = f y x
-                let rows = rows |> Array.map (c (-) 1)
-                let cols = cols |> Array.map (c (-) 1)
+            fun data ->
+                let rows = Array.zeroCreate data.Length
+                let cols = Array.zeroCreate data.Length
+                let values = Array.zeroCreate data.Length
+
+                Array.Parallel.iteri
+                    (fun i struct(packedIndex, value) ->
+                        let (rowIdx, columnIdx) = unpack packedIndex
+                        rows.[i] <- rowIdx
+                        cols.[i] <- columnIdx
+                        values.[i] <- value
+                    ) data
+
+                let flip f x y = f y x
+                let rows = rows |> Array.map (flip (-) 1)
+                let cols = cols |> Array.map (flip (-) 1)
                 {
                     Rows = rows
                     Columns = cols
                     Values = values
-                    RowCount = mtx.RowCount
-                    ColumnCount = mtx.ColumnCount
+                    RowCount = mtx.Shape.RowCount
+                    ColumnCount = mtx.Shape.ColumnCount
                 }
 
     let transposeCOO (matrix: COOFormat<'a>) =
-        (matrix.Rows, matrix.Columns, matrix.Values)
-        |||> Array.zip3
-        |> Array.sortBy (fun (row, col, value) -> col, row)
-        |> Array.unzip3
+        printfn "Start transpose COO"
+
+        (matrix.Columns, matrix.Rows, matrix.Values)
+        |||> Array.map3 (fun colIdx rowIdx value -> struct(pack colIdx rowIdx, value))
+        |> Array.sortBy (fun struct(p, _) -> p)
         |>
-            fun (rows, cols, values) ->
+            fun data ->
+                let rows = Array.zeroCreate data.Length
+                let cols = Array.zeroCreate data.Length
+                let values = Array.zeroCreate data.Length
+
+                Array.Parallel.iteri
+                    (fun i struct(packedIndex, value) ->
+                        let (rowIdx, columnIdx) = unpack packedIndex
+                        rows.[i] <- rowIdx
+                        cols.[i] <- columnIdx
+                        values.[i] <- value
+                    ) data
+
                 {
-                    Rows = cols
-                    Columns = rows
+                    Rows = rows
+                    Columns = cols
                     Values = values
                     RowCount = matrix.ColumnCount
                     ColumnCount = matrix.RowCount
                 }
+
+    let getMatricesFilenames configFilename =
+        let getFullPathToConfig filename =
+            Path.Combine [|
+                __SOURCE_DIRECTORY__
+                "Configs"
+                filename
+            |] |> Path.GetFullPath
+
+        configFilename
+        |> getFullPathToConfig
+        |> File.ReadAllLines
```

Comment:
What the reason to convert array to Seq? If you really want to work with Seq, then just use ```File.ReadLines``` instead of ```File.ReadAllLines```. 

But I'm not sure that Seq is necessary here. The configuration file is relatively small.